### ISSUE TITLE
Adapter does not honor columnName configuration in destroy callback

### ISSUE THREAD

**Issue by sgress454** [balderdashy/sails#3989]
_From @bossyang on November 7, 2015 3:52_

**Model**

``` javascript
    code: {
      type: 'string',
      required: true,
      columnName: 'sale_point_code'
    },
```

**Unit Test (Mocha)**

``` javascript
  it('should remove store', done => {
    let newStore = Fixture.newStore;
    Store
      .destroy({ code: newStore.code })
      .then(stores => {
        assert.equal(stores[0].code, newStore.code);
        done();
      })
      .catch(done);
  });
```

It will pass if change the assert express to stores[0].sale_point_code, i.e. the columnName.

```
models:Store should remove store:
     AssertionError: expected undefined to equal '530A'
      at test/unit/models/Store.test.js:44:16
      at tryCatcher (node_modules/waterline/node_modules/bluebird/js/main/util.js:26:23)
      at Promise._settlePromiseFromHandler (node_modules/waterline/node_modules/bluebird/js/main/promise.js:503:31)
      at Promise._settlePromiseAt (node_modules/waterline/node_modules/bluebird/js/main/promise.js:577:18)
      at Promise._settlePromises (node_modules/waterline/node_modules/bluebird/js/main/promise.js:693:14)
      at Async._drainQueue (node_modules/waterline/node_modules/bluebird/js/main/async.js:123:16)
      at Async._drainQueues (node_modules/waterline/node_modules/bluebird/js/main/async.js:133:10)
      at Immediate.Async.drainQueues [as _onImmediate] (node_modules/waterline/node_modules/bluebird/js/main/async.js:15:14)
```

_Copied from original issue: balderdashy/sails-mysql#260_

**Comment by sgress454** [comment]
_From @devinivy on November 30, 2015 14:54_

@bossyang can you add a failing test for this in [waterline-adapter-tests](https://github.com/balderdashy/waterline-adapter-tests)?

**Comment by sgress454** [comment]
This looks to be a Waterline issue that's been fixed in v0.13.x (for use with Sails 1.0), in both the `afterDestroy` callback and the records returned via the `fetch` metakey.  Confirmed still an issue with the 0.11.x version of Waterline.

**Comment by bossyang** [comment]
Sorry, I haven't used Sails.js for a quite while.
